pull_request.py

```diff
@@ -0,0 +1,138 @@
+import unidiff
+
+import util
+
+class PullRequest:
+  def __init__(self, repo, pr_number, target_commit, users):
+    self.repo_ = repo
+    self.pr_number_ = pr_number
+    self.target_commit_ = target_commit
+    self.users_ = users
+
+    self.pr_json_ = util.request(self.base_pr_url_()).json()
+
+    self.reviews = self.calculate_reviews_()
+    if self.author() in users:
+      self.reviews[self.author()] = 'APPROVED'
+
+    self.approvals = []
+    self.rejections = []
+    for user in users:
+      if user in self.reviews:
+        review = self.reviews[user]
+        if review == 'APPROVED':
+          self.approvals.append(user)
+        else:
+          self.rejections.append(user)
+
+    self.non_participants = [ user for user in users
+                              if user not in self.approvals
+                              and user not in self.rejections ]
+
+  def created_at_ts(self):
+    return util.iso8601_to_ts(self.pr_json_['created_at'])
+
+  def pushed_at_ts(self):
+    return util.iso8601_to_ts(self.pr_json_['head']['repo']['pushed_at'])
+
+  def last_changed_ts(self):
+    return max(self.created_at_ts(),
+               self.pushed_at_ts())
+
+  def days_since_created(self):
+    return util.days_since(self.created_at_ts())
+
+  def days_since_pushed(self):
+    return util.days_since(self.pushed_at_ts())
+
+  def days_since_changed(self):
+    return util.days_since(self.last_changed_ts())
+
+  def calculate_reviews_(self):
```

Comment:
I used trailing underscores for making private because I forgot the python convention; I think a final underscore is JS. Happy to switch. This should probably be a follow-up though.

Reply:
Sure, I'm ok with in a follow-up. (Slightly annoying that small PR changes are discouraged by approval system)

Reply:
I don't see why it needs to be a follow-up - our player base is more active at this point, getting re-approval doesn't seem like a big deal. Jeff is just trying to get extra points :P

Reply:
sure, whatever -- done in d27580e78f5827f73795790fce994407ec7be78b